ads/google/a4a/utils.js

```diff
@@ -243,9 +244,11 @@ export function groupAmpAdsByType(win, type, groupFn) {
  * @param {!Window} win
  * @param {!Node|!../../../src/service/ampdoc-impl.AmpDoc} nodeOrDoc
  * @param {number} startTime
+ * @param {string=} opt_ampAdImplementation
  * @return {!Promise<!Object<string,null|number|string>>}
  */
-export function googlePageParameters(win, nodeOrDoc, startTime) {
+export function googlePageParameters(
+  win, nodeOrDoc, startTime, opt_ampAdImplementation) {
```

Comment:
Remind me... does AMP not allowed setting default value for optional parameters:  opt_ampAdImplementation = AmpAdImplementation.AMP_AD_XHR_TO_IFRAME_OR_AMP.

I still think you should be consistent here and pass the a4a instance so that you can use a4a.isXhrAllowed()